---
title: "HUMAnN2"
output:
  html_document:
    theme: united
    fig_width: 5
    fig_height: 5
  pdf_document:
    fig_width: 5
    fig_height: 5
---
Back to [Table of Contents](index.html)  

**All of the code in this page is meant to be run on the command line unless otherwise specified.**
## Install Anaconda
Download and install [Anaconda3](https://www.continuum.io/downloads)

Linux: Right-click on the "Linux 64-bit" button under PYTHON 3.5. 
Mac: use the command-line installer or the graphical installer.

** on the command line:**
```{r eval=FALSE, engine='bash'}
# install Anaconda with python 3.5
bash Anaconda3-4.0.0-MacOSX-x86_64.sh

# now reopen your terminal window/login again

# now install a "python2" environment also containing anaconda
# This will let you switch between python2 and 3 easily
conda create -n humann2 python=2.7 anaconda

# enter the python 2 environment that you just created
source activate humann2

# exit with
source deactivate
```


## Install HUMAnN2

Note: Extensive documentation is [here](https://bitbucket.org/biobakery/humann2/src/tip/doc/UserManual.md?fileviewer=file-view-default).

1. Download it from [here](https://pypi.python.org/pypi/humann2) using `wget`.
2. Extract with `tar xvzf humann2-0.7.0.tar.gz` or whatever the filename is.
3. Enter the directory, then run the install script.

```{r eval=FALSE, engine='bash'}

# enter the humann2 environment (if you are not already in it)
source activate humann2

# Make sure you are inside the downloaded humann2 directory, then run:
python setup.py install
```

4. Download two databases that it needs
```{r eval=FALSE, engine='bash'}
# be sure to change the path ~/humann2_databases if desired
humann2_databases --download chocophlan full ~/humann2_databases
humann2_databases --download uniref uniref50_diamond ~/humann2_databases
```

## Install Metaphlan2
1. [Download it](https://bitbucket.org/biobakery/metaphlan2/get/default.zip) with `wget https://bitbucket.org/biobakery/metaphlan2/get/default.zip`.
2. Extract it (to the preferred installation directory, say `~/lib/biobakery-metaphlan2-c07beded1dc3`.
3. Add the appropriate paths to the system environment variables
```{r eval=FALSE, engine='bash'}
# Add these lines to ~/.bash_profile
export PATH=$HOME/lib/biobakery-metaphlan2-c07beded1dc3:$PATH
export mpa_dir=$HOME/lib/biobakery-metaphlan2-c07beded1dc3
```

## Run HUMAnN2

To run on a single fastq (or fasta) file:
```{r eval=FALSE, engine='bash'}
humann2 -i input.fastq -o output
```

To run on a directory  fastq (or fasta) file:
```{r eval=FALSE, engine='bash'}
# only use R1 reads
for f in dir/*_R1_*.fastq; echo $f; humann2 -i $f -o output --output-format biom --remove-stratified-output --output-max-decimals 0; done

# note: to run them in parallel (if you have many processors), replace
# the ";" before "done" with a "&"
# But do this with caution

# Then merge the files 
merge_otu_tables.py -i $SAMPLE1_genefamilies.biom,$SAMPLE2_genefamilies.biom,$SAMPLE3_genefamilies.biom -o genefamilies_all.biom
merge_otu_tables.py -i $SAMPLE1_pathabundance.biom,$SAMPLE2_pathabundance.biom,$SAMPLE3_pathabundance.biom -o pathabundance_all.biom
merge_otu_tables.py -i $SAMPLE1_pathcoverage.biom,$SAMPLE2_pathcoverage.biom,$SAMPLE3_pathcoverage.biom -o pathcoverage_all.biom

# convert to TSV if you want open as a spreadsheet
biom convert --to-tsv -i genefamilies.biom -o genefamilies.txt --table-type "Ortholog table"
biom convert --to-tsv -i pathcoverage.biom -o pathcoverage.txt --table-type "Pathway table"
biom convert --to-tsv -i pathabundance.biom -o pathabundance.txt --table-type "Pathway table"
```

